nominal/core/client.py

```diff
@@ -986,19 +1011,53 @@ def _logs_to_conjure(
 
 def _create_search_assets_query(
     search_text: str | None = None,
-    label: str | None = None,
+    labels: Sequence[str] | None = None,
     properties: Mapping[str, str] | None = None,
 ) -> scout_asset_api.SearchAssetsQuery:
     queries = []
     if search_text is not None:
-        q = scout_asset_api.SearchAssetsQuery(search_text=search_text)
-        queries.append(q)
-    if label is not None:
-        q = scout_asset_api.SearchAssetsQuery(label=label)
-        queries.append(q)
+        queries.append(scout_asset_api.SearchAssetsQuery(search_text=search_text))
+
+    if labels is not None:
+        for label in labels:
+            queries.append(scout_asset_api.SearchAssetsQuery(label=label))
 
     if properties:
         for name, value in properties.items():
             queries.append(scout_asset_api.SearchAssetsQuery(property=scout_run_api.Property(name=name, value=value)))
 
     return scout_asset_api.SearchAssetsQuery(and_=queries)
+
+
+def _handle_deprecated_labels_properties(
+    function_name: str,
+    label: str | None,
+    labels: Sequence[str] | None,
+    property: tuple[str, str] | None,
+    properties: Mapping[str, str] | None,
+) -> tuple[Sequence[str], Mapping[str, str]]:
+    if labels is None:
+        labels = []
+
+    # back-compat for existing code which used label
+    if label:
+        warnings.warn(
+            f"parameter 'label' of {function_name} is deprecated, use 'labels' instead",
+            UserWarning,
+            stacklevel=2,
+        )
+        labels = [label, *labels]
```

Comment:
May be worth ensuring that `label` and `labels` are not used together:

```
if all((label, labels)):
   raise
```

And same for `properties` below. A user that is aware of the new syntax should have no reason to use the old one, and should at the very least not mix them.